omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/AlertRecipientResource1_8.java

```diff
@@ -0,0 +1,190 @@
+import org.openmrs.module.webservices.rest.web.resource.impl.NeedsPaging;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.notification.Alert;
+import org.openmrs.notification.AlertRecipient;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Sub-resource for {@link AlertRecipient}
+ */
+@SubResource(parent = AlertResource1_8.class, path = "recipient", supportedClass = AlertRecipient.class, supportedOpenmrsVersions = {
+		"1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*", "2.2.*", "2.3.*", "2.4.*", "2.5.*" })
+public class AlertRecipientResource1_8 extends DelegatingSubResource<AlertRecipient, Alert, AlertResource1_8> {
+
+	private static final String RECIPIENT = "recipient";
+
+	private static final String ALERT_READ = "alertRead";
+
+	private static final String DATE_CHANGED = "dateChanged";
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("uuid");
+		description.addProperty("display");
+		description.addSelfLink();
+
+		if (rep instanceof DefaultRepresentation) {
+			description.addProperty(RECIPIENT, Representation.REF);
+			description.addProperty(ALERT_READ);
+			description.addProperty(DATE_CHANGED);
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+		} else if (rep instanceof FullRepresentation) {
+			description.addProperty(RECIPIENT, Representation.DEFAULT);
+			description.addProperty(ALERT_READ);
+			description.addProperty(DATE_CHANGED);
+		}
+
+		return description;
+	}
+
+	@PropertyGetter("display")
+	public String getDisplayProperty(AlertRecipient instance) {
+		return instance.getRecipient().getDisplayString();
+	}
+
+	@PropertySetter(RECIPIENT)
+	public void setRecipientProperty(AlertRecipient instance, User recipient) {
+		String recipientUuid = recipient.getUuid();
+		if (!StringUtils.isEmpty(recipientUuid)) {
+			User newRecipient = Context.getUserService().getUserByUuid(recipientUuid);
+			if (newRecipient != null) {
+				instance.setRecipient(newRecipient);
+			}
+		}
+	}
+
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty(RECIPIENT);
+		return description;
+	}
+
+	@Override
+	public Model getGETModel(Representation rep) {
+		ModelImpl modelImpl = ((ModelImpl) super.getGETModel(rep))
+				.property("uuid", new StringProperty())
+				.property("display", new StringProperty());
+
+		if (rep instanceof DefaultRepresentation) {
+			modelImpl
+					.property(RECIPIENT, new RefProperty("#/definitions/UserGetRef"))
```

Comment:
@BartlomiejRasztabiga  thanks for the work on this issue,
Could you fix the indenting here and in other places as done on line 62-69 above..

Reply:
@gitcliff Should be resolved now :)
I've kept all of them at one-tab indentation. I know that on GitHub it looks different than in IDE.